shared/desktop/remote/sync-browser-window.desktop.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+// This HOC wraps a component that represents a remote window. When this component is mounted anywhere it'll make a BrowserWindow
+import * as React from 'react'
+import electron from 'electron'
+import hotPath from '../hot-path'
+import menuHelper from '../app/menu-helper'
+import {injectReactQueryParams} from '../../util/dev'
+import {resolveRootAsURL} from '../resolve-root'
+import {showDevTools, skipSecondaryDevtools} from '../../local-debug.desktop'
+
+const BrowserWindow = electron.BrowserWindow || electron.remote.BrowserWindow
+const ipcRenderer = electron.ipcRenderer
+
+const devScripts = __DEV__
+  ? [
+      {
+        async: false,
+        src: resolveRootAsURL('dist', 'dll/dll.vendor.js'),
+      },
+      {
+        async: false,
+        src: hotPath('common-chunks.js'),
+      },
+    ]
+  : []
+
+type Props = {
+  windowOpts?: Object,
+  positionBottomRight?: boolean,
+  component: string,
+  title: string,
+  selectorParams: string,
+}
+
+const defaultWindowOpts = {
+  frame: false,
+  fullscreen: false,
+  height: 300,
+  resizable: false,
+  show: false, // Start hidden and show when we actually get props
+  width: 500,
+}
+
+type State = {
+  remoteWindow: ?BrowserWindow,
+}
+
+const sendLoad = (webContents: any, selectorParams: string, component: string, title: ?string) => {
+  webContents.send('load', {
+    component,
+    scripts: [
+      ...devScripts,
+      {
+        async: false,
+        src: hotPath('component-loader.bundle.js'),
+      },
+    ],
+    selectorParams,
+    title,
+  })
+}
+
+function BrowserWindowSync(ComposedComponent: any) {
```

Comment:
I think it would be helpful to easily understand interdependencies if this was typed:

Type is probably something like:

```
function BrowserWindowSync<P>(ComposeComponent: React.Component<{...P, remteWindow: BrowserWindow}>) {
...
```

Reply:
🐑  🐑  Also it might be nice to just pass down what we need right now, so it's a bit clearer what consumers of the prop can do. i.e. just pass down `remoteWindowEmit: (eventName, eventArgs) => void`

Reply:
but this hoc doesn't understand what its children needs. Yes, thats all the syncProps might need but this isn't true in general. this hoc is all about managing a BrowserWindow so thats what it passes down